subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/plugins/JavaGradlePluginPluginPublishingIntegrationTest.groovy

```diff
@@ -170,6 +171,40 @@ class JavaGradlePluginPluginPublishingIntegrationTest extends AbstractIntegratio
         ivyRepo.module('com.example.foo', 'com.example.foo' + PLUGIN_MARKER_SUFFIX, '1.0').assertPublished()
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/23551")
+    def "Can publish maven with changed artifactId"() {
+
+        given:
+        plugin('foo', 'com.example.foo')
+        publishToMaven()
+
+        and:
+        buildFile << """
+            publishing {
+                afterEvaluate {
+                    publications {
+                        getByName("pluginMaven"){
+                            configure{
+                                artifactId = "foo-new"
```

Comment:
To increase coverage, I would change all 3 things that are affected in the test.

Reply:
Sounds good. Test case updated.